### PR TITLE
Added info about mview

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
## Purpose of this pull request

This pull request (PR) extends information about indexing, adds information about mview.xml.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/extension-dev-guide/indexing.html
- https://devdocs.magento.com/guides/v2.2/extension-dev-guide/indexing.html

whatsnew
Added [information](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/indexing.html) about indexing and the `mview.xml` file.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/extension-dev-guide/indexing.md [hunk 1]
```diff
@@ -112,6 +112,73 @@ The following figure shows an example of setting indexers to Update by Schedule:
...
+    </view>
+<!-- ... -->
+```
+
+Explanation of nodes:
+* The `view` node defines an indexer. The `id` attribute is a name of the indexer table, the `class` attribute is indexer executor, the `group` attribute defines
+the indexer group. 
+* The `subscriptions` node is a list of tables for tracking changes.
+* The `table` node defines the certain table to observe and track changes. The attribute `name` is a name of an observable table, the attribute `entity_column` is an identifier column of entity to be re-indexed. So, in case of `catalog_category_product`, whenever one or more categories is saved, updated or deleted in `catalog_category_entity` the method
+execute of `Magento\Catalog\Model\Indexer\Category\Product` will be called with argument `ids` containing ids of entities from column defined
```
Comment (rogyar) [inline guides/v2.2/extension-dev-guide/indexing.md]:
```suggestion
the `execute` method of `Magento\Catalog\Model\Indexer\Category\Product` will be called with argument `ids` containing ids of entities from column defined
```

File: guides/v2.2/extension-dev-guide/indexing.md [hunk 2]
```diff
@@ -112,6 +112,73 @@ The following figure shows an example of setting indexers to Update by Schedule:
...
+under `entity_column` attribute. If indexer type is set to Update on Save the method being called right away after the operation if it set to Update by Schedule
```
Comment (rogyar) [inline guides/v2.2/extension-dev-guide/indexing.md]:
```suggestion
under `entity_column` attribute. If indexer type is set to "Update on Save", the method is called right away after the operation. If it's set to "Update by Schedule"...
```

File: guides/v2.2/extension-dev-guide/indexing.md [hunk 3]
```diff
@@ -112,6 +112,73 @@ The following figure shows an example of setting indexers to Update by Schedule:
...
+the mechanism creates a record in the change log table using MYSQL triggers.
+
+A change log table is created according to the naming rule - INDEXER_TABLE_NAME + '_cl', in case of `catalog_category_product` it will be `catalog_category_product_cl`.
+The table contains the `version_id` auto-increment column and `entity_id` column that contains identifiers of entities to be re-indexed.
+For each `table` node the framework automatically creates MYSQL AFTER triggers for each possible event (INSERT, UPDATE, DELETE).
+
+For the table `catalog_category_entity` will be created triggers with the following statements:
```
Comment (rogyar) [inline guides/v2.2/extension-dev-guide/indexing.md]:
```suggestion
For the table `catalog_category_entity` triggers will be created with the following statements.
```

File: guides/v2.2/extension-dev-guide/indexing.md [hunk 4]
```diff
@@ -112,6 +112,73 @@ The following figure shows an example of setting indexers to Update by Schedule:
...
+
+DELETE operation:
+```mysql
+BEGIN
+    INSERT IGNORE INTO `catalog_category_product_cl` (`entity_id`) VALUES (OLD.`entity_id`);
+END
+
+```
+
+The method `Magento\Framework\Mview\ViewInterface::update` responsible for handling records in the changelog. The method is being called by CRON and
```
Comment (rogyar) [inline guides/v2.2/extension-dev-guide/indexing.md]:
```suggestion
The method `Magento\Framework\Mview\ViewInterface::update` is responsible for handling records in the changelog. The method is being called by CRON and
```

File: guides/v2.2/extension-dev-guide/indexing.md [hunk 5]
```diff
@@ -112,6 +112,73 @@ The following figure shows an example of setting indexers to Update by Schedule:
...
+it defines ID's to be re-indexed from the change log by last applied `version_id` and call the method `execute` for each particular indexer with
```
Comment (rogyar) [inline guides/v2.2/extension-dev-guide/indexing.md]:
```suggestion
it defines IDs to be re-indexed from the change log by last applied `version_id` and calls the `execute` method for each particular indexer with IDs as an argument
```

**Comment by dobooth** [comment]
running tests

**Pull request merged**